client/executor/wasmtime/src/runtime.rs

```diff
@@ -130,8 +225,27 @@ fn create_compiled_unit(
 	// Instantiate and link the env module.
 	let global_exports = context.get_global_exports();
 	let compiler = new_compiler(compilation_strategy)?;
-	let env_module = instantiate_env_module(global_exports, compiler, host_functions)?;
-	context.name_instance("env".to_owned(), env_module);
+
+	let mut missing_functions_stubs = if allow_missing_func_imports {
+		scan_missing_functions(code, host_functions)?
+	} else {
+		// If there are in fact missing functions they will be detected at the instantiation time
+		// and the module will be rejected.
+		MissingFunctionStubs::new()
+	};
+
+	let env_missing_functions = missing_functions_stubs.stubs.remove("env").unwrap_or_else(|| Vec::new());
```

Comment:
Can we not do this in the loop below and just check:
```
if module == env {
    host_functions
} else {
    &[]
}
```

Reply:
The main point of doing that is not in host functions but rather in creating a new instance and giving it a name for `non-env` modules.